languages/SMW_LanguageAr.php

```diff
@@ -52,62 +54,58 @@ class SMWLanguageAr extends SMWLanguage {
 
 	protected $m_SpecialProperties = array(
 		// always start upper-case
-		'_TYPE'  => 'لديه نوع',
-		'_URI'   => 'معرف الموارد الموحد معادلة',
-		'_SUBP' => 'الخاصية الفرعية ل',
-		'_SUBC' => 'تصنيف فرعي من',
-		'_UNIT' => 'عرض الوحدات',
-		'_IMPO' => 'المستوردة من',
-		'_CONV' => 'يقابل',
-		'_SERV' => 'يوفر الخدمة',
-		'_PVAL' => 'يسمح بالقيمة',
+		'_TYPE' => 'من النوع',
+		'_URI'  => 'URL مكافئ',
+		'_SUBP' => 'خصيصة فرعية من',
+		'_SUBC' => 'مُصنّف على التصنيف',
+		'_UNIT' => 'وحدة العرض',
+		'_IMPO' => 'مستوردة من',
+		'_CONV' => 'توافق',
+		'_SERV' => 'تقدّم خدمة',
+		'_PVAL' => 'تقبل القيمة',
 		'_MDAT' => 'تاريخ التعديل',
-		'_CDAT' => 'Creation date', // TODO: translate
-		'_NEWP' => 'Is a new page', // TODO: translate
-		'_LEDT' => 'Last editor is', // TODO: translate
-		'_ERRP' => 'يمتلك قيمة غير صحيحة ل',
-		'_LIST' => 'يمتلك حقول',
-		'_SOBJ' => 'Has subobject', // TODO: translate
-		'_ASK'  => 'Has query', // TODO: translate
-		'_ASKST'=> 'Query string', // TODO: translate
-		'_ASKFO'=> 'Query format', // TODO: translate
-		'_ASKSI'=> 'Query size', // TODO: translate
-		'_ASKDE'=> 'Query depth', // TODO: translate
-		'_ASKDU'=> 'Query duration', // TODO: translate
-		'_MEDIA'=> 'Media type',
-		'_MIME' => 'MIME type',
-		'_ERRC' => 'Has processing error',
-		'_ERRT' => 'Has processing error text',
-		'_PREC'  => 'Display precision of',
-		'_LCODE' => 'Language code',
-		'_TEXT'  => 'Text',
-		'_PDESC' => 'Has property description',
-		'_PVAP'  => 'Allows pattern',
-		'_DTITLE' => 'Display title of',
-		'_PVUC' => 'Has uniqueness constraint',
+		'_CDAT' => 'تاريخ الإنشاء',
+		'_NEWP' => 'هي صفحة جديدة',
+		'_LEDT' => 'آخر مَن حرّرها',
+		'_ERRP' => 'فيها قيمة غير صحيحة في',//الخصيصة
```

Comment:
> Whether those changes are for the better or for the worse. This will remain controversial.

I agree here.
> I also know that the Arabic Wikipedia community has become increasingly more conservative; not only in regard to localisation.

Maybe you are right. However, Arabic Wikipedia is the largest Arabic language wiki on the Internet. So, the opinion of the Arabic Wikipedia community should be taken into consideration here. After all, Semantic MediaWiki is a MediaWiki extension and it was we (Arabic Wikipedia users) who started the Arabic MediaWiki translation in the first place.
>I know they are used in Gnome, a major FOSS package, as well as the Mozilla suite.

Both were translated by Arab Eyes people if I was not mistaken.
>The Arabic keyboard is evil, but that's another discussion.

I agree.
>For me if is is useful, it should be allowed; people should be given the chance to try them.

I respect your opinion but we disagree here. We simply don't "invent" language. In Wikipedia, we use only words and phrases commonly used in Arabic sources/literature.
>  that it is an essential part of the Arabic writing system

Yes, I know. Both of us have more than 10 years experience with Wikipedia. However, you only make a few edits every month while I do make hundreds of edits every week. That's why we disagree here. The issue of diacritics is very annoying when are trying to find a certain word in 150 kilobytes article/webpage and you don't have time to read all the article/webpage. I agree that web browsers and search engines code should be modified as you described. However until this happens, let's not make things harder for very active users (who make most edits in any wiki as you know).

Another note: You know well that most Arabic websites don't use diacritics for the reasons you and I stated here (due to the difficulties of presentation; i.e fonts, resolutions, encoding).
> This is all mostly of the past now.

Not in my opinion.
> prevents the use of concise expressions that are arguably more suitable in certain cases, for fear of ambiguity.

True. However, the problem of using Arabic diacritics has no ideal solution. We are just weighting pros and cons.

Bottom line: I have stated my opinion here and I have nothing further to add. I have already submitted a patch and the decision of whether to merge your patch or mine is up to kghbln. I was happy to have this conversation with you and thank you.